.github/workflows/workflow.yml

```diff
@@ -39,6 +39,33 @@ jobs:
           SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
       with:
           command: code test
-          
- 
+    
+  production:
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+
+    steps: 
+    - name: checkout
+      uses: actions/checkout@v2
 
+    - uses: satackey/action-docker-layer-caching@v0.0.11
+    
+    - name: Build, Push and Release a Docker container to Heroku
+      uses: gonuit/heroku-docker-deploy@v1.3.3
+      with:
+        email: ${{ secrets.HEROKU_EMAIL }}
+        heroku_api_key: ${{ secrets.HEROKU_API_KEY }}
+        heroku_app_name: ${{ secrets.HEROKU_APP_NAME }}
+        dockerfile_directory: ./
+        dockerfile_name: Dockerfile
+        process_type: web
+        
+    - name: docker login
+      run: docker login -u ${{ secrets.DOCKER_USERNAME }} -p ${{ secrets.DOCKER_PASSWORD }}
```

Comment:
Fairly minor here but you might notice this produce a warning: we should ideally action this step using the `--password-stdin` flag rather than passing the password explicitly, to avoid leaking those credentials to shell logs. [See the docs](https://docs.docker.com/engine/reference/commandline/login/) for more info, example of how that might look below:
```
echo ${{ secrets.DOCKER_PASSWORD }} | docker login -u ${{ secrets.DOCKER_USERNAME }} --password-stdin
```